Source/PartnerCenter.CustomerPortal/BusinessLogic/Commerce/PaymentGateways/PayUGateway.cs

```diff
@@ -0,0 +1,437 @@
+            myremotepost.SetUrl(this.GetPaymentUrl(payconfig.AccountType));
+            myremotepost.SetInputs(inputs);
+            return myremotepost;
+        }
+
+        /// <summary>
+        /// Throws PartnerDomainException by parsing PayUMoney exception. 
+        /// </summary>
+        /// <param name="ex">Exceptions from PayUMoney API call.</param>        
+        private void ParsePayUException(PartnerDomainException ex)
+        {
+            throw new PartnerDomainException(ErrorCode.PaymentGatewayFailure).AddDetail("ErrorMessage", ex.Message);
+        }
+
+        /// <summary>
+        /// Retrieves the Order from a payment transaction.
+        /// </summary>
+        /// <param name="operation">operation data.</param>
+        /// <param name="prod">product data.</param>
+        /// <param name="quant">quantity data.</param>
+        /// <returns>The Order for which payment was made.</returns>
+        private async Task<OrderViewModel> GetOrderDetails(string operation, string prod, string quant)
+        {
+            OrderViewModel orderFromPayment = null;
+            try
+            {
+                orderFromPayment = new OrderViewModel();
+                List<OrderSubscriptionItemViewModel> orderSubscriptions = new List<OrderSubscriptionItemViewModel>();
+
+                orderFromPayment.OperationType = (CommerceOperationType)Enum.Parse(typeof(CommerceOperationType), operation, true);
+                string[] prodList = prod.Split(':');
+                string[] quantList = quant.Split(':');
+
+                for (int i = 0; i < prodList.Length; i++)
+                {
+                    orderSubscriptions.Add(new OrderSubscriptionItemViewModel()
+                    {
+                        SubscriptionId = prodList[i],
+                        OfferId = prodList[i],
+                        Quantity = Convert.ToInt32(quantList[i], CultureInfo.InvariantCulture)
+                    });
+                }
+
+                orderFromPayment.Subscriptions = orderSubscriptions;
+            }
+            catch (PartnerDomainException ex)
+            {
+                this.ParsePayUException(ex);
+            }
+
+            return await Task.FromResult(orderFromPayment);
+        }
+
+        /// <summary>
+        /// Throws PartnerDomainException by parsing PayUMoney exception. 
+        /// </summary>
+        /// <returns>return payment configuration</returns>
+        private async Task<PaymentConfiguration> GetAPaymentConfigAsync()
+        {
+            // Before getAPIContext ... set up PayUMoney configuration. This is an expensive call which can benefit from caching. 
+            PaymentConfiguration paymentConfig = await ApplicationDomain.Instance.PaymentConfigurationRepository.RetrieveAsync();
+
+            return paymentConfig;
+        }
+
+        /// <summary>
+        /// Remote post class.
+        /// </summary>
+        public class RemotePost
```

Comment:
>public class RemotePost [](start = 8, length = 23)

This is a public class and only used by PayUGateway. If you intend to keep this public then this should live in a separate class file. Else mark this as a private class.

Reply:
Done.

---
In reply to: [183243478](https://github.com/PartnerCenterSamples/Reseller-Web-Application/pull/61#discussion_r183243478) [](ancestors = 183243478)